flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManagerImpl.java

```diff
@@ -153,6 +155,8 @@ public SlotManagerImpl(
 		this.waitResultConsumedBeforeRelease = waitResultConsumedBeforeRelease;
 		this.defaultWorkerResourceSpec = defaultWorkerResourceSpec;
 		this.numSlotsPerWorker = numSlotsPerWorker;
+		this.defaultSlotResourceProfile = defaultWorkerResourceSpec != null ?
+			generateDefaultSlotResourceProfile(defaultWorkerResourceSpec, numSlotsPerWorker) : null;
```

Comment:
Instead of setting `defaultSlotResourceProfile` to `null`, can't we set it to `ResourceProfile.UNKNOWN` or something like this?